### PR TITLE
Only add `facility_not_found` error if facility is not found. 

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Updates API's `record-utils` controller to not add `sys.facility_not_found` error by default.
Updates `update_clinics` to add `sys.facility_not_found` error when facility is not found. 
Updates `update_clinics` to not update contact when matching by `refid`. 
Refactors `update_clinics` to be only use promises. 

medic/medic#3543

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Please have a look, @garethbowen . Thanks!

File: shared-libs/transitions/src/lib/utils.js [hunk 1]
```diff
@@ -313,5 +314,7 @@ module.exports = {
   isNonEmptyString: expr => typeof expr === 'string' && expr.trim() !== '',
   evalExpression: (expr, context) => vm.runInNewContext(expr, context),
 
-  getSubjectIds: contact => registrationUtils.getSubjectIds(contact)
+  getSubjectIds: contact => registrationUtils.getSubjectIds(contact),
+
+  XFORM_CONTENT_TYPE
```
Comment (garethbowen) [inline shared-libs/transitions/src/lib/utils.js]: I think instead of exposing a string it'd be more useful to expose a function, eg:

```js
isXFormReport: doc => doc && doc.content_type === 'xml' 
```

It gives more context about how it should be used and removes some duplicate code too!

File: shared-libs/transitions/src/transitions/update_clinics.js [hunk 2]
```diff
@@ -1,107 +1,63 @@
...
+      .then(data => {
+        if (!data.rows.length || !data.rows[0].id) {
+          return;
+        }
+
+        return lineage.fetchHydratedDoc(data.rows[0].id);
+      });
   }
+
+  return Promise.resolve();
```
Comment (garethbowen) [inline shared-libs/transitions/src/transitions/update_clinics.js]: I find this quite hard to follow, I think because there's a lot going on in each branch of the `if`/`else`. Consider splitting each branch out as an individual function, eg:

```js
const getContact = doc => {
  if (doc.refid) {
    return getContactByRefId(...);
  }
  if (doc.from) {
    return getContactByPhone(...);
  }
  return Promise.resolve();
};
```

File: shared-libs/transitions/src/lib/utils.js (hunk 1 above)
Comment (dianabarsan) [reply shared-libs/transitions/src/lib/utils.js]: Done!

File: shared-libs/transitions/src/transitions/update_clinics.js (hunk 2 above)
Comment (dianabarsan) [reply shared-libs/transitions/src/transitions/update_clinics.js]: Done!

**Comment by dianabarsan** [comment]
I've rebased to resolve conflicts and to be ready to publish the lib in correct condition once AT-ed.
